pkg/network/ebpf/c/protocols/tls/go-tls-location.h

```diff
@@ -101,17 +104,18 @@ static __always_inline int read_register(struct pt_regs* ctx, int64_t regnum, vo
         *(u64*)dest = tmp;
         return 0;
     #elif defined(__TARGET_ARCH_arm64)
-        // TODO Support ARM
-        /*if (regnum >= 0 && regnum < sizeof(ctx->regs)) {
-            // Verifier won't allow direct access to regs array if the index is not const
-            switch (regnum) {
-                case 0:
-                    __builtin_memcpy(dest, &ctx->regs[0], sizeof(ctx->regs[0]));
-                default:
-                    return 1;
+        volatile u64 tmp = 0;
+    #pragma unroll
+        for (int i = 0; i < NUM_REGISTERS; i++) {
+            if (i == regnum) {
+                tmp = ctx->regs[i];
             }
+        }
+
+        *(u64*)dest = tmp;
+        if (regnum < NUM_REGISTERS) {
             return 0;
-        }*/
+        }
         return 1;
```

Comment:
if think the check should be the first do be done, as the iteration is redundant in such case
```c
if (regnum >= NUM_REGISTERS) {
    return 1;
}
```

Reply:
Agreed